playbooks/tasks/portal-configs-load.yml

```diff
@@ -100,6 +100,12 @@
       {% endfor %}
   failed_when: mongodb_config_server_preference_result.failed | default(False)
 
+# If portal_modules is defined in hosts.ini, update the values
+- name: Update portal_modules
+  set_fact:
+    webportal_server_config: "{{ webportal_server_config | combine({'portal_modules': portal_modules}, recursive=True) }}"
```

Comment:
What is wanted behavior here?

I thought we wanted this:
- if portal modules are set for specific server in LastPass, we use exactly this (not `combine`)
- otherwise we use value from `hosts.ini`

I other words: portal modules are defined in `hosts.ini` with optional override from server config.
If we use `combine` here we set portal modules to union of the above configs.

Reply:
the goal here is that we are consistent between LastPass and hosts.ini

so i want to override the field. so should I not have `recursive=True` here?

Reply:
Looking once more on this.

I think we shouldn't save configs from `hosts.ini` back to LastPass (via `webportal_server_config`).

Current process is (without `hosts.ini` configs):

- load server config(s) from LastPass
- set default values if the vars are not set in webportal_server_config (LastPass)
- save back (missing vars) to server config in LastPass

With `hosts.ini` I think it should be something like:

- load server config(s) from LastPass
- if portal_modules is not defined in `hosts.ini` and portal_modules is not defined in `webportal_server_config` then set default value in `webportal_server_config`
- portal_modules = portal_modules | default(webportal_server_config.portal_modules)
- set default values if the vars are not set in webportal_server_config (but exclude those set in `hosts.ini`)
- save back (missing vars) to server config in LastPass

If we move more and more config vars to `hosts.ini` we could divide `webportal_server_config_defaults` to something like:
- `webportal_server_config_defaults`
- `webportal_server_config_defaults_if_not_in_hosts_ini`

where the first works as usually (sets the default value if missing, to be saved to LastPass)
and the second sets the default value (to be saved to LastPass) only if missing in hosts.ini

Reply:
Summary from Discord Call.

Flows
1) not defined in hosts.ini or LastPass
 - this is error case, fail

2) not defined in hosts.ini, defined in LastPass
 - current state, preserve backwards compatibility
 - load from lastpass into `webportal_server_config.portal_modules`
 - Save back to lastpass 

3) defined in hosts.ini and defined in LastPass
 - override LastPass value on load with hosts.ini value
 - set LastPass value to `Deprecated: moved to hosts.ini` on save

4) defined in hosts.ini and not defined in LastPass
 - new ideal state
 - add check if `webportal_server_config.portal_modules is defined` before setting to `portal_modules`
 - nothing saved back to lastpass 

For `.env` file use `PORTAL_MODULES=portal_modules | default(webportal_server_config.portal_modules)`